### DISCUSSION TITLE
Understanding Kubernetes setup within the Akash Network

### DISCUSSION THREAD

**Discussion by Be-P** [akash-network/community#171] [Q&A]
Hi everyone, I'm trying to understand how does the Akash Network integrate Kubernetes. 

How are the Kubernetes Control Plane "master nodes" are elected and managed?
How does networking work within the context of Kubernetes on Akash?

**Comment by cloud-j-luna** [comment]
Hello @Be-P !

Each Provider runs its own Kubernetes cluster. It behaves as a normal instance wit CRDs, Operators and services that make part of the Akash Network Provider solution. There are no network "master nodes". Every provider is independent from one another by default and as of this time there is no native High Availability setup between Providers.

**Comment by shimpa1** [comment]
hi @Be-P !
Akash software runs on top of a Kubernetes cluster, The Kubernetes master node is used as a control plane for the cluster. There can be at least 1 control node in a cluster but there can also be many, depending on the size of the cluster. All the control nodes use a quorum mechanism to determine the leader node. For a functional control-plane (which consists of all the control nodes) half the control nodes +1 need to be healthy (n/2+1). This means if your cluster has 3 control nodes, at least 2 control nodes need to be healthy in order to elect a new leader node. This is all standard Kubernetes stuff, nothing Akash-specific.
The networking in Akash is done using Calico as the CNI. In case that the provider is offering IP leases, Metallb is used additionally.
Inter-Pod communication is disabled for security purposes. The Akash Ingress Controller pod is used for external communication over ports 80/443.
